tests/container_checker/test_container_checker.py

```diff
@@ -0,0 +1,259 @@
+    restarted. If the container was not restarted by the command 'config reload', then we
+    start it and check its status.
+
+    Args:
+        duthost: Hostname of DUT.
+        stopped_container_list: names of stopped containers.
+
+    Returns:
+        None.
+    """
+    for container_name in stopped_container_list:
+        logger.info("Checking the running status of container '{}'".format(container_name))
+        if is_container_running(duthost, container_name):
+            logger.info("Container '{}' is running.".format(container_name))
+        else:
+            logger.info("Container '{}' is not running and restart it...".format(container_name))
+            duthost.shell("sudo systemctl restart {}".format(container_name))
+            logger.info("Waiting until container '{}' is restarted...".format(container_name))
+            restarted = wait_until(CONTAINER_RESTART_THRESHOLD_SECS,
+                                   CONTAINER_CHECK_INTERVAL_SECS,
+                                   check_container_state, duthost, container_name, True)
+            if not restarted:
+                if is_hiting_start_limit(duthost, container_name):
+                    clear_failed_flag_and_restart(duthost, container_name)
+                else:
+                    pytest.fail("Failed to restart container '{}'".format(container_name))
+
+            logger.info("Container '{}' was restarted".format(container_name))
+
+
+def test_container_checker(duthosts, rand_one_dut_hostname, tbinfo):
+    """Tests the feature of container checker.
+
+    This function will check whether the container names will appear in the Monit
+    alerting message if they are stopped explicitly or they hit start limitation.
+
+    Args:
+        duthosts: list of DUTs.
+        rand_one_dut_hostname: hostname of DUT.
+        tbinfo: Testbed information.
+
+    Returns:
+        None.
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+    container_autorestart_states = duthost.get_container_autorestart_states()
+    disabled_containers = get_disabled_container_list(duthost)
+
+    bgp_neighbors = duthost.get_bgp_neighbors()
+    up_bgp_neighbors = [ k.lower() for k, v in bgp_neighbors.items() if v["state"] == "established" ]
+
+    skip_containers = disabled_containers[:]
+    # Skip 'radv' container on devices whose role is not T0.
+    if tbinfo["topo"]["type"] != "t0":
+        skip_containers.append("radv")
+
+    stopped_container_list = stop_containers(duthost, container_autorestart_states, skip_containers)
+    pytest_assert(len(stopped_container_list) > 0, "None of containers was stopped!")
+
+    # Wait for 6 minutes such that Monit has a chance to write alerting message into syslog.
+    logger.info("Sleep 6 minutes to wait for the alerting message...")
+    time.sleep(360)
+
+    check_alerting_message(duthost, stopped_container_list)
+
+    logger.info("Executing the config reload...")
+    config_reload(duthost)
```

Comment:
@bingwang-ms @wangxin Please help me review this PR again. The previous PR (https://github.com/Azure/sonic-mgmt/pull/2852) was reverted due to BGP sessions were down after all containers are restarted and then it failed the post-check on KVM testbed. The root reason is I should not restart the containers according to the sequence of stopped containers. The correct way is to use `config_reload(...)` to restart the containers since some containers are depended on others.  After running the `config_reload(...)`, the line 255 will check whether all stopped containers are running or not and it will restart some containers in case the `config_reload(...)` did not restart them such as the container `mgmt-framework`.

Reply:
LGTM. Except for the ```time.sleep(360)``` that guohan has mentioned.

Reply:
Thanks, Bing! Please help me review the function `update_monit_service(...)` and the line 274.

Reply:
Why do we have to do config_reload here? is config_reload_after_tests not enough?

Reply:
This script will first do the `config_reload(...)` to start the stopped containers and then do the post check to check whether BGP session are established and all critical processes are running,

Reply:
Why not move config_reload and the checking below to config_reload_after_tests?

Reply:
OK, Will move the post checking after the `config_reload(...)` in the fixture.